src/NRules/NRules.Fluent/RuleTypeScanner.cs

```diff
@@ -134,6 +141,19 @@ public IRuleTypeScanner Type(params Type[] types)
         return this;
     }
 
+    /// <summary>
+    /// Filters rule types using a predicate.
+    /// </summary>
+    /// <param name="predicate">The filter to use.</param>
+    /// <returns>Rule type scanner to continue scanning specification.</returns>
+    public IRuleTypeScanner Where(Func<Type, bool> predicate)
+    {
+        var ruleTypesToKeep = _ruleTypes.Where(predicate);
```

Comment:
I think the chained `Where` + `Except` + `Contains` Linq calls are going to be doing multiple iterations over the entire `_ruleTypes` list for each `Contains` call in `RemoveAll`. 

Recommend optimizing and simplifying this with just `_ruleTypes.RemoveAll(x => !predicate(x));`

But i'm also wondering if storing the delegate and applying it in that way (if not null) in `GetRuleTypes` rather than mutating the list may be more consistent with the existing private and nested type filters?